compiler/modules/types/core.bal

```diff
@@ -22,11 +22,11 @@ public type UniformTypeCode
     |UT_XML_RW|UT_LIST_RW|UT_MAPPING_RW|UT_TABLE_RW|UT_OBJECT_RW
     |UT_STREAM|UT_FUTURE;
 
-type Atom RecAtom|TypeAtom;
+public type Atom RecAtom|TypeAtom;
```

Comment:
Better for types to provide some abstractions to support this: e.g. isAtomRecursive function.

Reply:
fixed